src/cc/lib/distributed/client.cc

```diff
@@ -418,8 +418,8 @@ void GRPCClient::GetNodeFeature(std::span<const NodeId> node_ids, std::span<Feat
             auto curr_feature_reply = reply.feature_values().c_str();
             for (auto index : reply.offsets())
             {
-                std::copy(curr_feature_reply, curr_feature_reply + fv_size, curr_feature_out + fv_size * index);
-                curr_feature_reply += fv_size;
+                std::copy(curr_feature_reply + fv_size * index, curr_feature_reply + fv_size * (index + 1),
```

Comment:
That looks wrong, `reply.offsets()` can't be always increasing.. Previously we were sequentially copying from `reply.feature_values()`, but you changed it to a random offset..